modules/gmc-agent/src/main/java/org/gridgain/service/ActionService.java

```diff
@@ -98,9 +116,24 @@ private Response convertToErrorResponse(UUID id, Throwable e) {
         log.error(String.format("Failed to execute action, send error response to GMC: [reqId=%s]", id), e);
 
         return new Response()
-                .setStatus(FAILED)
                 .setId(id)
-                .setError(new ResponseError(INTERNAL_ERROR_CODE, e.getMessage(), e.getStackTrace()));
+                .setStatus(FAILED)
+                .setError(new ResponseError(getErrorCode(e), e.getMessage(), e.getStackTrace()));
+    }
+
+    /**
+     * @param e Exception.
+     * @return Integer error code.
+     */
+    private int getErrorCode(Throwable e) {
```

Comment:
How about to use enum instead of error codes? Does it make sense?

Reply:
We are using https://www.jsonrpc.org/specification#error_object spec, I think we can use enum instead of integers, because we don't support more thank 1 argument in methods and bulk requests.